js/ftx.js

```diff
@@ -1121,6 +1121,65 @@ module.exports = class ftx extends Exchange {
         return this.parseBalance (result);
     }
 
+    async fetchAllBalances (params = {}) {
+        await this.loadMarkets ();
+        const response = await this.privateGetWalletAllBalances (params);
+        //
+        // {
+        //     "success": true,
+        //     "result": {
+        //       "main": [
+        //         {
+        //           "coin": "USDTBEAR",
+        //           "free": 2320.2,
+        //           "spotBorrow": 0.0,
+        //           "total": 2340.2,
+        //           "usdValue": 2340.2,
+        //           "availableWithoutBorrow": 2320.2
+        //         },
+        //         {
+        //           "coin": "BTC",
+        //           "free": 2.0,
+        //           "spotBorrow": 0.0,
+        //           "total": 3.2,
+        //           "usdValue": 23456.7,
+        //           "availableWithoutBorrow": 2.0
+        //         }
+        //       ],
+        //       "Battle Royale": [
+        //         {
+        //           "coin": "USD",
+        //           "free": 2000.0,
+        //           "spotBorrow": 0.0,
+        //           "total": 2200.0,
+        //           "usdValue": 2200.0,
+        //           "availableWithoutBorrow": 2000.0
+        //         }
+        //       ]
+        //     }
+        // }
+        //
+        const returnObject = {
+            'info': response,
+        };
+        const data = this.safeValue (response, 'result', []);
+        const accounts = Object.keys (data);
+        for (let i = 0; i < accounts.length; i++) {
+            const accountName = accounts[i];
+            const balances = data[accountName];
+            returnObject[accountName] = {};
+            for (let j = 0; j < balances.length; j++) {
+                const balance = balances[j];
+                const code = this.safeCurrencyCode (this.safeString (balance, 'coin'));
+                const account = this.account ();
+                account['free'] = this.safeString2 (balance, 'availableWithoutBorrow', 'free');
+                account['total'] = this.safeString (balance, 'total');
+                returnObject[accountName][code] = account;
```

Comment:
Logically, we have to do the following here:

```JavaScript
        balance[code] = account;
    }
    result[accountId] = parseBalance (balance);
}
return result;
```

Just like we do here:

- https://github.com/ccxt/ccxt/blob/master/js/huobi.js#L1977
- https://github.com/ccxt/ccxt/blob/master/js/okex3.js#L1640

Reply:
@kroitor balance is an object from the api response, so it doesn't make sense to parse that object, I'm not exactly sure what you're trying to describe?

Reply:
@kroitor I'm just waiting on an answer here

Reply:
@samgermain i'll explain it in more details asap

Reply:
@kroitor Do you have any more details on this?

Reply:
@samgermain in your initial output the values of `free`, `used`, `total` are all undefined. I think `parseBalance` has to be applied to each of the accounts and the entire fetchAllBalances method should return parsed numbers.